### ISSUE TITLE
CSName parameters should (sometimes?) expand protected macros

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#1006]
Minimal latex example:
```latex
\documentclass{article}
\usepackage{etoolbox}
\begin{document}

\expandafter\def\csname test-def\endcsname{prints the word def}
\expandafter\def\csname test-robust\endcsname{prints the word robust}

\def\mydef{def}
\newrobustcmd\myrobust{robust}

\csname test-\mydef\expandafter\endcsname

\csname test-\myrobust\expandafter\endcsname

\end{document}
```

The PDF result is:
```
prints the word def
prints the word robust
```

The latexml result is correct for the def case, but fails with an error in the robust case:
```
Error:unexpected:\myrobust The control sequence \myrobust should not appear between \csname and \endcsname
```

Tracing this down, I landed in `State::lookupExpandable` where this peculiar line is responsible for the discrepancy:
```perl
    && ($toplevel || !$$defn{isProtected})) { # is this the right logic here? don't expand unless digesting?
```

Simply commenting out that condition gets the example working, and fixes the general issue of using robust macros in expandable contexts. 

However, running a `make test` shows failures in various places, so I'm assuming the real adjustment is more subtle.

**Comment by dginev** [comment]
Alternatively, maybe the `CSName` parameter type should expand with the `$toplevel` flag set to true? Would need to read into the documentation to really know what's ajar.

**Comment by dginev** [comment]
FWIW, switching the toplevel flag to true and ensuring it's passed to the State method keeps all tests passing. I will add my commits to the etoolbox PR, but feel free to offer feedback in this issue - trying to avoid clutter.

Side-note, I had mentioned the `\protected` mechanism print in `\meaning` via email to Bruce and I'll also add my tweak to that to the PR.

**Comment by dginev** [comment]
All pushed.

https://github.com/brucemiller/LaTeXML/pull/978/files#diff-0bc08f4db903eb28daf537319b493b85R847

**Comment by dginev** [comment]
Solved by merging #978
